### ISSUE TITLE
'Adam' object has no attribute 'get_weights' 

### ISSUE THREAD

**Issue by ddjamalova** [nadavbra/protein_bert#46]
I ran the finetuning on Google Colab and it returned this error message
```
AttributeError                            Traceback (most recent call last)
[<ipython-input-12-5e6f22672ea3>](https://localhost:8080/#) in <module>
----> 1 finetune(model_generator, input_encoder, OUTPUT_SPEC, train_set['seq'], train_set['label'], valid_set['seq'], valid_set['label'], \
      2         seq_len = 512, batch_size = 32, max_epochs_per_stage = 5, lr = 1e-04, begin_with_frozen_pretrained_layers = True, \
      3         lr_with_frozen_pretrained_layers = 1e-02, n_final_epochs = 1, final_seq_len = 1024, final_lr = 1e-05, callbacks = training_callbacks)

2 frames
[/usr/local/lib/python3.9/dist-packages/proteinbert/model_generation.py](https://localhost:8080/#) in update_state(self, model)
     33     def update_state(self, model):
     34         self.model_weights = copy_weights(model.get_weights())
---> 35         self.optimizer_weights = copy_weights(model.optimizer.get_weights())
     36 
     37     def _init_weights(self, model):

AttributeError: 'Adam' object has no attribute 'get_weights'
```
I tried to change the optimizer by model_generator.compile but it didn't work out. 
Could you suggest to me how can I fix this error, please?

**Comment by avivlazar** [comment]
Hi, 
I'm not sure if its the best solution, but works for me (in Google Colab): installed old tensorflow version:
`!pip install tensorflow==2.9.0`
Note: Same issue was posted in StackOverflow: 
https://stackoverflow.com/questions/74761791/attributeerror-adam-object-has-no-attribute-get-weights

If there is another solution, I'll be glad to hear.
Regards,
Aviv.

**Comment by nadavbra** [comment]
ProteinBERT was written for a version of tensorflow from over a year ago and there might be small code changes required to make it compatible with newer versions (I expect it to be pretty small changes). @ddofer is leading an effort to update ProteinBERT at some point in the future.

**Comment by r-kellerm** [comment]
@ddjamalova , @avivlazar - please note that this issue is already fixed in the latest code ([PR #44](https://github.com/nadavbra/protein_bert/pull/44))
